pkgs/applications/editors/jupyter-kernels/iruby/default.nix

```diff
@@ -2,16 +2,41 @@
 , bundlerApp
 }:
 
-bundlerApp {
-  pname = "iruby";
-  gemdir = ./.;
-  exes = [ "iruby" ];
+# Jupyter console:
+# nix run --impure --expr 'with import <nixpkgs> {}; jupyter-console.withSingleKernel iruby.definition'
 
-  meta = with lib; {
-    description = "Ruby kernel for Jupyter";
-    homepage    = "https://github.com/SciRuby/iruby";
-    license     = licenses.mit;
-    maintainers = [ maintainers.costrouc ];
-    platforms   = platforms.unix;
+# Jupyter notebook:
+# nix run --impure --expr 'with import <nixpkgs> {}; jupyter.override { definitions = { iruby = iruby.definition; }; }'
+
+let
+  app = bundlerApp {
+    pname = "iruby";
+    gemdir = ./.;
+    exes = [ "iruby" ];
+
+    meta = {
+      description = "Ruby kernel for Jupyter";
+      homepage    = "https://github.com/SciRuby/iruby";
+      license     = lib.licenses.mit;
+      maintainers = with lib.maintainers; [ costrouc thomasjm ];
+      platforms   = lib.platforms.unix;
+    };
+  };
+
+in
+
+app.overrideAttrs (oldAttrs: {
```

Comment:
bundlerApp does not passthru passthru? If so please fix that in bundleApp instead

Reply:
It does...the reason I structured it this way is because the passthru needs to call the `app` itself (`"${app}/bin/iruby"`). Is there a better way to do that?

Reply:
we normally solve that by assigning the builder command to a variable in a let in and then use that straight in there without an override

Reply:
Resolved with your suggestions